### PR TITLE
[WIP] rcS replace old tone_alarm usage with tune_control

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

**Comment by dagar** [comment]
There's still more to fix.

**Comment by santiago3dr** [comment]
flash v4, getting error tones upon boot

**Comment by santiago3dr** [comment]
flashed v2, getting same error tones, solid lite Blue and error messages on qgroundcontrol
errors on qgc

**Comment by dagar** [comment]
Blocking boot in many situations.

**Comment by dagar** [comment]
@davids5 @simonegu looks like tone_alarm stm32 is pretty out of sync with kinetis and samv7.

**Comment by hamishwillee** [comment]
@dagar There is no documentation anywhere of tune_control (or the system it replaces). Personally I'd like every module, driver, whatever have at least the kind of high level docs we find in https://github.com/PX4/Firmware/blob/master/src/templates/module/module.cpp ... and allow them to be auto extracted.

Thoughts?

**Comment by dagar** [comment]
If we switch tune_control to use ModuleBase then we should get the generated documentation.

**Comment by bkueng** [comment]
`tune_control` is just a command, so you don't need to use ModuleBase. But we should replace https://github.com/PX4/Firmware/blob/master/src/systemcmds/tune_control/tune_control.cpp#L65, with the `PRINT_MODULE_*` methods.

**Comment by hamishwillee** [comment]
So will `PRINT_MODULE_*` allow `PX4_INFO` to also output usage on command line?

**Comment by bkueng** [comment]
> So will PRINT_MODULE_* allow PX4_INFO to also output usage on command line and automatically get picked up in docs?

Yes exactly. You use it instead of `PX4_INFO`

File: src/systemcmds/tune_control/tune_control.cpp [hunk 1]
```diff
@@ -120,7 +121,18 @@ tune_control_main(int argc, char *argv[])
 			break;
 
 		case 't':
-			value = (uint8_t)(strtol(myoptarg, nullptr, 0));
+
+			tune_string = myoptarg;
+
+			if (!strcmp(tune_string, "startup")) {
```
Comment (simonegu) [inline src/systemcmds/tune_control/tune_control.cpp]: :+1:

File: src/lib/tunes/default_tunes.cpp [hunk 2]
```diff
@@ -1,61 +0,0 @@
...
- ****************************************************************************/
-
-/**
- * @file default_tunes.h
- */
-
-#include "tunes.h"
-
-// initialise default tunes
-const char *Tunes::_default_tunes[] = {
```
Comment (simonegu) [inline src/lib/tunes/default_tunes.cpp]: @dagar this file was created with the purpose of having the possibility on having custom tunes even for already present ones without the need to modify the library code, basically making it independent from the library itself.

File: ROMFS/px4fmu_common/init.d/rcS [hunk 3]
```diff
@@ -110,7 +105,7 @@ then
 	tone_alarm start
 
 	# play startup tone
-	tune_control play -t 1
+	tune_control play -t startup
```
Comment (simonegu) [inline ROMFS/px4fmu_common/init.d/rcS]: nice :smile:

File: src/lib/tunes/default_tunes.cpp (hunk 2 above)
Comment (dagar) [reply src/lib/tunes/default_tunes.cpp]: Ok, I'll leave it as is.

**Pull request closed without merging**